### PR TITLE
New package: qemu

### PR DESCRIPTION
**Pull request by anderbubble**
Introducing a qemu package. This is my first new spack package, so comments welcome.

I had a little trouble declaring a dependency:

```
depends_on('pixman@0.21.8:')
```

but this tried to install 0.21.8 rather than the preferred version that is at or newer than 0.21.8. For now I've dropped the version specifier, but there's probably something I'm missing.

I'm also using `preferred=True`; but it might be better to just mark the `-rc` versions as `@develop` or something similar. I couldn't figure out how to do that, so let me know.

### CODE REVIEW

File: var/spack/repos/builtin/packages/qemu/package.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+# them, you can save this file and test your package like this:
+#
+#     spack install qemu
+#
+# You can edit this file again by typing:
+#
+#     spack edit qemu
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qemu/package.py]: Remove this fixme stuff

File: var/spack/repos/builtin/packages/qemu/package.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+from spack import *
+
+
+class Qemu(AutotoolsPackage):
+    """QEMU is a generic and open source machine emulator and virtualizer."""
+
+    # FIXME: Add a proper url for your package's homepage here.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qemu/package.py]: Remove this line

File: var/spack/repos/builtin/packages/qemu/package.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    homepage = "https://www.qemu.org"
+    url      = "https://download.qemu.org/qemu-4.1.1.tar.xz"
+
+    # FIXME: Add a list of GitHub accounts to
+    # notify when the package is updated.
+    # maintainers = ['github_user1', 'github_user2']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qemu/package.py]: Remove this if you don't want to add yourself as a maintainer

File: var/spack/repos/builtin/packages/qemu/package.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+    version('4.2.0-rc3',  sha256='5d5aece78673c758f648b3ae06ea9758ffe0345b7db07d92f3e5ce9bdcd55ed8')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qemu/package.py]: You don't actually need to add every single version if you don't want to. I usually only add the latest stable release, and I rarely add release candidates.

File: var/spack/repos/builtin/packages/qemu/package.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    version('0.10.1',     sha256='632b8942d8c85b36997ce3ed893bc34c868b432fbaadd4ea86994ca881d6665b')
+    version('0.10.0',     sha256='cd554729fa9d0ec17164afbc1cea62d02bde3db8e16db3fd1b8e71d8e1b3dd41')
+    version('0.9.1',      sha256='a9655a471d0649f5540b890447b35849c162d9b986bf2bbddcb68461748e0f42')
+
+    # FIXME: Add dependencies if required.
+    # glib-2.40 gthread-2.0 is required to compile QEMU
+    depends_on('glib@2.40:')
+    depends_on('pixman') # software reports requirement >= 0.21.8
+
+    def configure_args(self):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qemu/package.py]: Remove this if you don't need it.

File: var/spack/repos/builtin/packages/qemu/package.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    version('0.10.3',     sha256='4a5393ff47493a80d28a53ce307bffa52a066fd82045a5598b8da074a2f7dfe7')
+    version('0.10.2',     sha256='2d62bcad5ef7609813381e77a6422fec3d4cbd6347f4297b687510ab4d32e813')
+    version('0.10.1',     sha256='632b8942d8c85b36997ce3ed893bc34c868b432fbaadd4ea86994ca881d6665b')
+    version('0.10.0',     sha256='cd554729fa9d0ec17164afbc1cea62d02bde3db8e16db3fd1b8e71d8e1b3dd41')
+    version('0.9.1',      sha256='a9655a471d0649f5540b890447b35849c162d9b986bf2bbddcb68461748e0f42')
+
+    # FIXME: Add dependencies if required.
+    # glib-2.40 gthread-2.0 is required to compile QEMU
+    depends_on('glib@2.40:')
+    depends_on('pixman') # software reports requirement >= 0.21.8
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qemu/package.py]:
```suggestion
    depends_on('pixman@0.21.8:')
```
should work fine, if not there's a bug.

**Comment by anderbubble** [comment]
PR updated with recommended changes.

- FIXME bits removed
- Maintainer contact added
- Copyright updated
- Dependency version specified
- RC versions removed

Beyond that, I understand that it's not required to have all versions in the package; but having run against missing versions in existing packages before, I see no reason to not include them.

File: var/spack/repos/builtin/packages/qemu/package.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+class Qemu(AutotoolsPackage):
+    """QEMU is a generic and open source machine emulator and virtualizer."""
+
+    homepage = "https://www.qemu.org"
+    url      = "https://download.qemu.org/qemu-4.1.1.tar.xz"
+
+    maintainers = ['anderbubble']
+
+    version('4.1.1',      sha256='ed6fdbbdd272611446ff8036991e9b9f04a2ab2e3ffa9e79f3bab0eb9a95a1d2', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qemu/package.py]: You don’t need preferred=True now that there aren’t any rc versions.

**Comment by adamjstewart** [comment]
@anderbubble did you accidentally close this PR?

**Comment by anderbubble** [comment]
@adamjstewart Yes. I accidentally deleted the branch rather than overwriting it. That same branch name is back up, _and_ it's rebased on top of current develop. II was going to open a new PR; but if you can reopen this one, that should work too. (I can't: it's greyed out or me.)

**Comment by anderbubble** [comment]
I went ahead and opened a new PR, #14184.

**Pull request closed without merging**